### DISCUSSION TITLE
Workflow for dealing with documentation updates

### DISCUSSION THREAD

**Discussion by porres** [pure-data/pddp#260] [General]
Hi, as pointed by @Lucarda in https://github.com/pure-data/pddp/issues/252 there is some noise and confusion about the workflow. The readme here says the readme tells people to contribute to a branch I don't have anymore and I should update it. But as discussed over there, the real target should be https://github.com/pure-data/pure-data/tree/documentation which is way behind master

I abandoned that as I was just no able to sync with master and had to @umlaeute for it and I also asked multiple times granting me powers to sync it.

Now I just got a message from @millerpuckette asking for a good way for him to make changes to the docs and I think that working on this https://github.com/pure-data/pure-data/tree/documentation branch should be the easy solution. I just can't sync it yet. Hopefully Miller can.

I would like once again to ask to be able to sync the documentation branch with master as it eventually gets quite necessary. Let me also point to this discussion from here on, when I was making several attempts to be able to do that and it all failed https://github.com/pure-data/pure-data/pull/2291#issuecomment-2103901173 the one thing that always always always gets in the way is a 'no-merge-commits-policy'

I already questioned that policy and I can also quotie @Spacechild1 , who agreed and wrote
  
"if @umlaeute managed to do this, you should be too! Did he explain to you what he did?

But if the no-merge-commits policy turns out to be too restrictive and awkward, we should probably lift it."

Anyway. Let's see if Miller can sync and work on this documentation branch and if we all agree that is the best idea. Then I'd like to fix the readme and that we could work this syncing issues out.

cheers

**Comment by millerpuckette** [comment]
I don't know what it would mean to 'sync' the documentation branch...  but as far as I can tell it's so far out of date the best thing would be to just delete it and continue doing what we've been doing before (Alexandre maintaining a separate repo that I can merge from).  I don't know why it didn't work for that branch to live on the pure-data github repo but it doesn't seem to be a problem to have it over on his own repo anyway).

Meanwhile, the whole git thing seems only to work when there aren't two people trying to modify the same file.  While Alexandre was busy basically rewriting every file in doc/ there was no way anyone else could document their changes to Pd without causing merge failures.  But now that things are in pretty good shape I guess it won't be necessary to make changes to 100s of files anymore, so it probably won't be a problem if I start to make changes (and, yes, 
document them :)

**Comment by porres** [comment]
> I don't know what it would mean to 'sync' the documentation branch...

I meant syncing up with the master branch, bringing it up to date with the commits in master.

> but as far as I can tell it's so far out of date the best thing would be to just delete it

Or sync it :) it should've been just as easy to sync and bring it up to date

And well, I see now that it has been deleted anyway ;)

> Meanwhile, the whole git thing seems only to work when there aren't
> two people trying to modify the same file. 
> While Alexandre was busy basically rewriting every file in doc/ there
> was no way anyone else could document their changes to Pd without
> causing merge failures.

The way to prevent it was actually working everybody together on the documentation branch. It seems like the best and simplest solution. It can be a nightmare if people are working on the same file on different branches and independent PRs. Yup, conflicts arise.

But I had also noted that to you earlier that I'm not that busy with it anymore and can just hold back until you work on it on master. I also mentioned this could be considered the simplest, but also not that sophisticated solution :) 

I am happy doing documentation work on my own fork and sending PRs, but if more people are to contribute and we need to streamline the workflow, we need a documentation branch in the pure-data repository where we all can deal with.
